### PR TITLE
Remove deprecated input options from MeshBuilder

### PR DESCRIPTION
**Pull request by pbauman**
And cleaned up a few other code smells.

### CODE REVIEW

File: src/solver/src/mesh_builder.C [hunk 1]
```diff
@@ -98,20 +81,18 @@ namespace GRINS
       }
 
     // Create UnstructuredMesh object (defaults to dimension 1).
-    libMesh::UnstructuredMesh* mesh;
+    std::shared_ptr<libMesh::UnstructuredMesh> mesh;
```
Comment (roystgnr) [inline src/solver/src/mesh_builder.C:84]: Why shared_ptr and not unique_ptr?  We'd expect to have one clear "owner" of the mesh throughout its lifetime, no?

Comment (pbauman) [reply src/solver/src/mesh_builder.C:84]: Yeah, it's an atavism. It should be a unique_ptr. I think when this code was written originally, there was no UniquePtr wrapper in libMesh, but there was a SharedPtr wrapper that later got converted to std::shared_ptr. I think. It was a while ago.

Comment (pbauman) [reply src/solver/src/mesh_builder.C:84]: I didn't change it now because that will touch other pieces of code. Was trying to keep the PR self-contained.

**Review by roystgnr — APPROVED** [review]
Still curious about that old design decision, but the new changes here look like a nice cleanup.

**Pull request merged**